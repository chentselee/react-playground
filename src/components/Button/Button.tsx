import React from "react";

type Props = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;

const Button: React.FC<Props> = ({ children, ...props }) => {
  return (
    <button
      {...props}
      className="uppercase px-3 py-0.5 border border-gray-600 rounded-sm hover:bg-gray-600 hover:text-gray-50 transition-colors"
    >
      {children}
    </button>
  );
};

export default Button;
