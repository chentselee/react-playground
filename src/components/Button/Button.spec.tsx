import { render } from "@testing-library/react";

import Button from "./index";

it("render", () => {
  const { container } = render(<Button />);

  expect(container).toMatchSnapshot();
});
