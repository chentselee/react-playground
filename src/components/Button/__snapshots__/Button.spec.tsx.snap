// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div>
  <button
    class="uppercase px-3 py-0.5 border border-gray-600 rounded-sm hover:bg-gray-600 hover:text-gray-50 transition-colors"
  />
</div>
`;
